### DISCUSSION TITLE
Integrate with Hono server

### DISCUSSION THREAD

**Discussion by matheusfernd** [Danilqa/node-file-router#67] [Q&A]
Congratulations on the project; I thought the route structure was excellent.

I want to know if this can be integrated with the Hono server in any way.

**Comment by Danilqa** [comment]
@matheusfernd Hi! Thank you very much! 

Are you going to use it on top of Node.js or some other environments? I use the Node.js API in my implementation, unfortunately: working with files, processes, etc. It might work in some environments with Node.js compatibility. 

For example, like this:
https://github.com/Danilqa/node-file-router/pull/68

However it doesn't work yet because of `fs.stat`

**Reply by matheusfernd** [reply]
Thanks for the example, I did the tests using Bun with the Hono server and it worked perfectly both in production and development, I just needed to make a small simple adjustment, as it seems that the adapter option `getMethod: (context) => context.req. method,` when using the Hono context, the returned method is capitalized.

https://github.com/Danilqa/node-file-router/blob/b0c30e88794d1debf2cfe7cf7051f049eb84a45d/src/file-router.ts#L58

Here you can put `method?.toLowerCase() === routeMethod?.toLowerCase())`

**Reply by Danilqa** [reply]
Exactly, thanks!
